hikey.mk

```diff
@@ -1,12 +1,35 @@
+################################################################################
+# User-defined variables
+# Edit so these match your target
+# NOTE: If making changes after a build, please clean before rebuilding!
+################################################################################
+# Non-secure user mode (root fs binaries): 32 or 64-bit [default 64]
 NSU ?= 64
-SU ?= 32
+# Secure kernel (OP-TEE OS): 32 or 64-bit [default 64]
 SK ?= 64
+# Secure user mode (Trusted Apps): 32 or 64-bit [default 32, requires SK=64 for 64]
+SU ?= 32
+
+# Console UARTs: 3 or 0
```

Comment:
Ok. So here's the plan:
1. Submit PR to optee_os to make default UART 3 instead of 0.
2. 
 ```
 make
 # Building with both NW and SW at UART3 (default)
 
 make CFG_HIKEY_UART=0
 # Building with NW (EDK2 and LINUX) UART3 and SW (ARM-TF and optee_os) at UART0
 ```
3.
 ```
 CFG_HIKEY_UART ?= 3
 ARM_TF_CONSOLE_UART ?= $CFG_HIKEY_UART
 OPTEE_CONSOLE_UART ?= $CFG_HIKEY_UART
 EDK2_CONSOLE_UART ?= 3
 LINUX_CONSOLE_UART ?= 3
 ```
 The different ARM_..., OPTEE_..., EDK2_... and LINUX_... will be moved down to the section about the component itself.

This approach will 'hard' code the normal side to UART3, but the more advanced users can still configure each component to the UART of their choosing if they wish.

Any objections?